Domaca01/src/Domaca01.jl

```diff
@@ -0,0 +1,187 @@
+module Domaca01
+using LinearAlgebra, Domaca01, Plots
+
+"""
+    A = (V, I)
+
+Za razpršene matrike definiramo novo strukturo RazprsenaMatrika, ki je hranjena v matrikah `V` in `I`.
+Matrika zaradi prostorskih zahtev hrani vrednosti v matrikah `V` in `I`, velikosti n x m. Pri tem velja:
+`V(i,j) = A(i, I(i, j))`.
+"""
+struct RazprsenaMatrika
+    V
+    I
+end
+
+"""
+    ix = getIndex(A, i, j)
+
+Vrne vrednost, ki se v matriki `A` nahaja na mestu `i, j`.
+"""
+function getIndex(A::RazprsenaMatrika, i, j)
+    idx = findfirst(isequal(j), A.I[i])
+    return idx === nothing ? 0.0 : A.V[i][idx]
+end
+Base.getindex(A::RazprsenaMatrika, i, j) = getIndex(A, i, j)
+
+"""
+    ix = setIndex(A, x, i, j)
+
+Na mesto `i, j` v matriki `A` vnese vrednost `x`.
+"""
+function setIndex(A::RazprsenaMatrika, i, j, e)
+    idx = findfirst(isequal(j), A.I[i])
+    if idx === nothing
+        ix = searchsortedlast(A.I[i], j)
+        insert!(A.I[i], ix + 1, j)
+        insert!(A.V[i], ix + 1, e)
+    else
+        A.V[i][idx] = e
+    end
+end
+Base.setindex!(A::RazprsenaMatrika, e, i, j) = setIndex(A, i, j, e)
+
+"""
+    first = firstIndex(A, i)
+
+Vrne prvo vrednost v matriki `A`.
+"""
+function firstIndex(A::RazprsenaMatrika, i)
+    return A.I[i][1]
+end
+Base.firstindex(A::RazprsenaMatrika, i) = firstIndex(A, i)
+
+
+"""
+    first = lastIndex(A, i)
+
+Vrne zanjo vrednost v matriki `A`.
+"""
+function lastIndex(A::RazprsenaMatrika, i)
+    return A.I[i][end]
+end
+Base.lastindex(A::RazprsenaMatrika, i) = lastIndex(A, i)
+
+"""
+    b = A  * x
+
+Izvede množenje matrike `A` in vektorja `x` iz desne.
+"""
+function Base.:*(A::RazprsenaMatrika, x)
+    n = length(A.V)
+    b = zeros(Float64, n)
+
+    for i in 1:n
+        for (j, idx) in enumerate(A.I[i])
+            b[i] += A.V[i][j] * x[idx]
+        end
+    end
+    return b
+end
+
+
+"""
+    x, iter = SOR(A, b, x0, omega, tol)
+
+Izračunaj vrednost `x` in število iteracij `iter` za matriko A s SOR iteracijo.
+Pri tem podamo vektor `b`, začetni približek `x0`, parameter SOR iteracije `omega` in
+pogoj za zaključek `tol`. Nekatere vrednosti omege ne konvergirajo vedno, zato je število
+iteracija omejeno na 1000.
+"""
+function SOR(A::RazprsenaMatrika, b, x0, omega, tol=1e-10)
+    n = length(b)
+    x = copy(x0)
+
+    iter = 0
+
+    while true
+        x_old = copy(x)
+        for i in 1:n
+            sum1 = 0.0
+            sum2 = 0.0
+
+            for j in A.I[i]
+                if j < i
+                    sum1 += A[i, j] * x[j]
+                elseif j > i
+                    sum2 += A[i, j] * x_old[j]
+                end
+            end
+
+            x[i] = (1 - omega) * x_old[i] + (omega / A[i, i]) * (b[i] - sum1 - sum2)
+        end
+
+        iter += 1
+
+        if maximum(abs.(A * x - b)) <= tol
+            break
+        end
+
+        if iter >= 1000
+            break
+        end
+    end
+
+    return x, iter
+end
+
+"""
```

Comment:
Podati morate tudi vozlišča grafa, ki jih fiksirate. Funckcija bi morala poleg matrike vrniti še desne strani. 

Na primer za graf s štirimi vozlišči `1, 2, 3, 4`, povezavami `(1, 2), (2, 3), (3, 4)` in od katerih sta prvo in zadnje vozlišče fiksirana v točkah (-4, -5) in (2, 3) , dobimo 2 sistema enačb za koordinate vozlišč `2` in `3`

```math
\begin{array}{ccccc}
-2x_2 &+& x_3 &=& -x_1 = 4\\
x_2 &-& 2x_3 &=& -x_4 = -2
\end{array}\quad
\begin{array}{ccccc}
-2y_2 &+& y_3 &=& -y_1 = 5\\
y_2 &-& 2y_3 &=& -y_4 = -3
\end{array}
```
Tako bi funkcija vrnila `2x2` matriko in dva vektorja desnih strani(za vsako koordinato 1 vektor):

```math
A = \begin{pmatrix}
-2 &1\\
1 & -2
\end{pmatrix}
```
in 

```math
b_x = [4, -2]\quad b_y=[5, -3]
```

Koordinate vozlišč `2` in `3` so tako rešitve sistemov $Ax = b_x$ za $x$ koordinate in $Ax=b_y$ za $y$ koordinate.